### ISSUE TITLE
sidekick containers with no hosted data volumes deleted on server reboot

### ISSUE THREAD

**Issue by juanluisbaptiste** [rancher/rancher#6395]
**Rancher Version:**

1.1.4

**Docker Version:**

1.12.2

**OS and where are the hosts located? (cloud, bare metal, etc):**

Baremetal, Ubuntu 16.04.1

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

Single node, internalDB, one rancher server and several machines with the agent installed.

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle.

**Steps to Reproduce:**

I have a service that has two containers along with their sidekick containers, one for the database and the other for the application. The images are [these ones](https://github.com/juanluisbaptiste/docker-otrs), with this docker-compose file:

```
otrs:
  image: juanluisbaptiste/otrs:4.0.17-otrs-4_0_x-b6f93cc3
  container_name: "otrs"
  expose:
  - "80"
  links:
  - mariadb:mariadb
  - postfix:postfix
  labels:
    io.rancher.scheduler.affinity:host_label: deploy.tag.otrs=otrs
    io.rancher.sidekicks: storage-otrs
  environment:
      OTRS_ROOT_PASSWORD: xxxxxxxx
      OTRS_DB_PASSWORD: xxxxxxx
      MYSQL_ROOT_PASSWORD: xxxxxx
      OTRS_POSTMASTER_FETCH_TIME: 0
#     OTRS_TICKET_COUNTER: 6649
      OTRS_NUMBER_GENERATOR: "Date"
      OTRS_HOSTNAME: support.xxxx.com
      OTRS_ADMIN_EMAIL: admin@xxxx.com
      OTRS_ORGANIZATION: "ACME"
  volumes_from:
   - storage-otrs
mariadb:
  image: juanluisbaptiste/otrs-mariadb:latest
  expose:
  - "3306"
  environment:
    MYSQL_ROOT_PASSWORD: xxxxx
  volumes_from:
   - storage-db
  labels:
    io.rancher.scheduler.affinity:host_label: deploy.tag.otrs=otrs
    io.rancher.sidekicks: storage-db
storage-db:
  image: centos/mariadb:latest
  volumes:
    - /var/lib/mysql
    - '/etc/localtime:/etc/localtime:ro'
#  command: /bin/true
  labels:
    io.rancher.scheduler.affinity:host_label: deploy.tag.otrs=otrs
    io.rancher.container.start_once: "true"
storage-otrs:
  image: juanluisbaptiste/otrs:latest
  volumes:
    - /opt/otrs
    - '/data/otrs/backups:/var/otrs/backups'
    - '/etc/localtime:/etc/localtime:ro'
  labels:
    io.rancher.scheduler.affinity:host_label: deploy.tag.otrs=otrs
    io.rancher.container.start_once: "true"
  command: /bin/true
postfix:
  image: juanluisbaptiste/postfix:latest
  expose:
  - "25"
  environment:
    SERVER_HOSTNAME: support.xxxx.com
    SMTP_PASSWORD: xxxx
    SMTP_SERVER: smtp.xxxx.com
    SMTP_USERNAME: otrs@xxxx.com
  volumes:
   - "/etc/localtime:/etc/localtime:ro"

```

After bringing up the containers I login and do some modification, like creating a new ticket.

**Results:** 

When the server where the container is hosted is rebooted and containers come back up, the database is in its initial state (no new ticket), instead of having the same content before the reboot. This only happens after a reboot, a stop/start from the service from within rancher works as expected.

**Expected:**

When the server is rebooted and containers come backup the data containers should have the content from before the reboot, but it's gone. This is how I have handled my containers in the past with docker-compose before starting moving to rancher. The same docker-compose file from above without the labels and with _restart: always_ on the otrs and mariadb containers has respected my data containers.

Obviously I could use a host mounted data container and have the data outside the containers, but with the mentioned OTRS image is tricky as the configuration directory is populated during image build, which means that I can't simply map it to an outside directory because the mount would hide all configurations.

So I'm missing something on my docker-compose file or is this an expected behavior ?

**Comment by deniseschannon** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.
